sdk/nodejs/pubsub/zMixins.ts

```diff
@@ -0,0 +1,65 @@
+// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+import * as pulumi from "@pulumi/pulumi";
+import { Topic } from "./topic";
+import * as cloudfunctions from "../cloudfunctions";
+
+export interface TopicEventCallbackFunctionArgs extends cloudfunctions.CallbackFunctionArgs {
+    callback?: TopicEventHandler;
+    callbackFactory?: () => TopicEventHandler;
+    httpsTriggerUrl?: never;
+    triggerHttp?: never;
+}
+
+export interface TopicMessagePublishedArgs {
+    failurePolicy?: cloudfunctions.FailurePolicy;
+}
+
+export interface TopicEventArgs {
+    triggerType: "publish"
+    failurePolicy?: cloudfunctions.FailurePolicy;
+}
+
+export interface TopicEvent {
+    "@type": "type.googleapis.com/google.pubsub.v1.PubsubMessage",
+
+    /**
+     * key/value pairs included with the topic even.
+     */
+    attributes: Record<string, string>;
+
+    /**
+     * Base64 encoded data.  Use `Buffer.from(pubSubMessage.data, 'base64')` to get raw bytes of the
+     * message.
+     */
+    data: string;
+}
+
+export type TopicEventHandler = cloudfunctions.Callback<TopicEvent, void>;
+
+declare module "./topic" {
+    interface Topic {
+        onMessagePublished(name: string, handler: TopicEventHandler, args?: TopicMessagePublishedArgs, opts?: pulumi.ComponentResourceOptions): cloudfunctions.CallbackFunction;
+        onMessageEvent(name: string, handler: TopicEventHandler, args: TopicEventArgs, opts?: pulumi.ComponentResourceOptions): cloudfunctions.CallbackFunction;

```

Comment:
Are these two the same?  Why two?  I only see one mentioned in docs:

> There is a single Cloud Pub/Sub event used by Cloud Functions, and it has the trigger type value google.pubsub.topic.publish.

Reply:
so, our general pattern has always been to have specific APIs, which often bubble down to a general 'onEvent' call.  That way if someone wants to hook up "let me know if *anything* happens".  This is useful, for example, in the Bucket case where tehre are many distinct object events that can happen.

In the case of only one sort of thing... i can see the argument we wouldn't do this.  Let me know what you think.

Reply:
have removed the second event.  we could always add it later if necessary.